app/dashboard/route.ts

```diff
@@ -4,10 +4,11 @@ import { inject as service } from '@ember/service';
 export default class Dashboard extends Route.extend({
     session: service('session'),
 
-    async beforeModel(...args) {
-        await this._super(...args);
+    async beforeModel(transition) {
+        await this._super(transition);
 
         if (!this.get('session.isAuthenticated')) {
+            transition.abort();
```

Comment:
Out of curiosity, do you have a sense of how much time it adds to start a whole new transition?